src/config.rs

```diff
@@ -263,37 +319,106 @@ pub async fn remappings() -> Result<(), ConfigError> {
-        Err(err) => {
-            return Err(err);
+pub async fn remappings_foundry(
+    dependency: &Dependency,
+    config_file: &str,
+    soldeer_config: SoldeerConfig,
+) -> Result<(), ConfigError> {
+    let contents = read_file_to_string(config_file);
+    let mut doc: DocumentMut = contents.parse::<DocumentMut>().expect("invalid doc");
+
+    let mut dependencies: Vec<Dependency> = vec![];
+    if soldeer_config.reg_remappings {
+        dependencies = match read_config_deps(config_file).await {
+            Ok(deps) => deps,
+            Err(err) => {
+                return Err(err);
+            }
+        };
+    }
+
+    if *dependency != Dependency::default() {
+        Paint::green(&format!(
+            "Adding dependency {}-{} to the remappings",
+            dependency.name, dependency.version
+        ));
+        dependencies.push(dependency.clone());
+    } else if !dependencies.is_empty() {
+        Paint::green("Regenerating all the remappings");
+    }
+
+    if dependencies.is_empty() {
+        return Ok(());
+    }
+
+    let profiles: &mut EditTable = doc["profile"].as_table_mut().unwrap();
+
+    for (_, profile) in profiles.iter_mut() {
+        if soldeer_config.reg_remappings || !profile.as_table().unwrap().contains_key("remappings")
+        {
+            profile["remappings"] = Item::Value(Value::Array(Array::new()));
+        }
+        for dep in &dependencies {
+            match add_dependency_to_remap(profile, dep) {
+                Ok(_) => {}
+                Err(err) => {
+                    return Err(err);
+                }
+            };
         }
-    };
-    if filename.contains("foundry.toml") {
-        return Ok(vec![true]);
     }
-    let contents: String = read_file_to_string(&filename);
 
-    // reading the contents into a data structure using toml::from_str
-    let data: Foundry = match toml::from_str(&contents) {
-        Ok(d) => d,
-        Err(_) => {
-            println!(
-                "{}",
-                Paint::yellow(&"The remappings field not found in the soldeer.toml and no foundry config file found or the foundry.toml does not contain the `[dependencies]` field.\nThe foundry.toml file should contain the `[dependencies]` field if you want to use it as a config file. If you want to use the soldeer.toml file, please add the `[remappings]` field to it with the `enabled` key set to `true` or `false`.\nMore info on https://github.com/mario-eth/soldeer\nThe installation was successful but the remappings feature was skipped.".to_string())
-            );
-            return Ok(vec![false]);
+    let mut file: std::fs::File =
+        fs::OpenOptions::new().write(true).append(false).truncate(true).open(config_file).unwrap();
+    if let Err(e) = write!(file, "{}", doc) {
+        eprintln!("Couldn't write to the config file: {}", e);
+    }
+
+    Ok(())
+}
+
+fn add_dependency_to_remap(profile: &mut Item, dependency: &Dependency) -> Result<(), ConfigError> {
```

Comment:
lots of `unwrap` in this function.

For example, instead of doing `if profile_table.contains_key("remappings") { ... }` you can do `if let Some(remappings) = profile_table.get("remappings") { ... }` and that spares you one unwrap.